### PR TITLE
Cli register appmesh

### PR DESCRIPTION
**Pull request by marcogschmidt**
BOT NOTES: 
resolves https://github.com/solo-io/supergloo/issues/214

### CODE REVIEW

File: cli/pkg/cmd/register/app_mesh.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+						Namespace: opts.RegisterAppMesh.ConfigMap.Namespace,
+						Name:      opts.RegisterAppMesh.ConfigMap.Name,
+					}
+				}
+			}
+
+			_, err := clients.MustMeshClient().Write(mesh, skclients.WriteOpts{})
+			if err != nil {
+				return errors.Wrapf(err, "failed to create mesh %v", mesh.Metadata.String())
+			}
```
Comment (EItanya) [inline cli/pkg/cmd/register/app_mesh.go]: The convention that we're following in other commands which create resources is to print the resource once it has been completed check install for example

File: cli/pkg/cmd/register/app_mesh_test.go [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+	})
+
+	Describe("flag validation works as expected", func() {
+
+		It("fails if no name was provided", func() {
+			err := utils.Supergloo(fmt.Sprintf(
+				"register appmesh --region %s --secret %s --auto-inject %s",
+				"us-east-1",
+				"DEFAULT.SECRET",
+				"false",
```
Comment (EItanya) [inline cli/pkg/cmd/register/app_mesh_test.go]: Did you figure out the weird behavior with bool flags

File: cli/pkg/options/values.go [hunk 3]
```diff
@@ -29,7 +29,7 @@ func (v *ResourceRefsValue) String() string {
 
 func (v *ResourceRefsValue) Set(s string) error {
 	split := strings.SplitN(s, ".", 2)
-	if len(split) != 2 {
+	if len(split) != 2 || strings.Trim(split[0], " ") == "" || strings.Trim(split[1], " ") == "" {
```
Comment (EItanya) [inline cli/pkg/options/values.go]: good call

File: cli/pkg/surveyutils/register_appmesh.go [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+	default:
+		return errors.Errorf("%v is not a known selector type", injectionSelectorType)
+	}
+
+	return nil
+}
+
+func surveyConfigMap(opts *options.Options) error {
+	var useCustomConfigMap string
+	if err := cliutil.ChooseFromList(
```
Comment (EItanya) [inline cli/pkg/surveyutils/register_appmesh.go]: There's boilerplate code for selecting resources, the function is called `surveyResources` in that package

File: cli/pkg/cmd/register/app_mesh_test.go (hunk 2 above)
Comment (marcogschmidt) [reply cli/pkg/cmd/register/app_mesh_test.go]: Yeah, it does not work with the other commands either, it's a bug. I filed [this issue](https://github.com/solo-io/supergloo/issues/323) to track it.

File: cli/pkg/cmd/register/app_mesh.go (hunk 1 above)
Comment (marcogschmidt) [reply cli/pkg/cmd/register/app_mesh.go]: Right, I forgot. Thank for catching that!

File: cli/pkg/options/values.go (hunk 3 above)
Comment (marcogschmidt) [reply cli/pkg/options/values.go]: 👍

File: cli/pkg/cmd/register/app_mesh_test.go (hunk 2 above)
Comment (EItanya) [reply cli/pkg/cmd/register/app_mesh_test.go]: okay cool

File: cli/pkg/surveyutils/register_appmesh.go (hunk 4 above)
Comment (marcogschmidt) [reply cli/pkg/surveyutils/register_appmesh.go]: Unfortunately I can't use it here as it accepts only our custom resources.

**Pull request merged**